*** old/mq/src/share/java/com/sun/messaging/jmq/jmsclient/MessageProducerImpl.java	2010-03-24 03:45:16.000000000 +0800
--- new/mq/src/share/java/com/sun/messaging/jmq/jmsclient/MessageProducerImpl.java	2010-10-11 16:55:43.000000000 +0800
***************
*** 77,82 ****
--- 77,83 ----
  
  public class MessageProducerImpl implements MessageProducer {
      protected boolean isClosed = false;
+     protected boolean isClosing = false;
      protected boolean disableMessageId = false;  //default in spec.
      protected boolean disableMessageTimestamp = false; //default in spec.
      protected int deliveryMode = DeliveryMode.PERSISTENT; //default in spec.
***************
*** 529,534 ****
--- 530,537 ----
      //need to address this for 2.1
      public void
      close() throws JMSException {
+         isClosing = true;
+ 
          //XXX PROTOCOL3.5 --
          // Producer flow control. Raptor broker keeps track of
          // producers.
***************
*** 1050,1056 ****
          }
  
          private synchronized void checkFlowControl(Message message) {
!             while (flowLimit == 0 && ! isClosed) {
                  try {
                      blocked = true;
                      wait();
--- 1053,1059 ----
          }
  
          private synchronized void checkFlowControl(Message message) {
!             while (flowLimit == 0 && ! isClosing) {
                  try {
                      blocked = true;
                      wait();
